File: files/regex/regex-demo.pl
#!/usr/bin/env perl
use strict;
use warnings;

# Perl Quick Regex Tutorial:
#   http://perldoc.perl.org/perlrequick.html
# Perl Long Regex Tutorial:
#   http://perldoc.perl.org/perlretut.html
# Perl Full Regex Syntax:
#   http://perldoc.perl.org/perlrequick.html

my $data = "chr1   65436543   rsID776";

# Regex to test if string matches
if ($data =~ m/^chr/) {
	print "Found match! line starts with 'chr'\n";
}

# Regex to extract information (ie. 'grouping')
# Get the number following the 'chr', store it in '$1'
if ($data =~ m/^chr([0-9]+)/) {
	print "found chromosome: $1\n";
}

File: files/regex/regex-demo.js
#!/usr/bin/env node

// Javascript Regex Tutorial:
//   https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
// Javascript Regex Reference:

var data = "chr1   65436543   rsID776";

// Test if 'data' matches the regex. returns NULL if no match found.
if (data.match(/^chr/)) {
    console.log("Found match! line starts with 'chr'");
}

// Regex to extract information (ie. 'grouping')
// Get the number following the 'chr', store it in array 'm'
// 'm' will be NULL if no match is found.
var m = data.match(/^chr([0-9]+)/);
if (m) {
   console.log("found chromosome: ", m[1]);
}
